TestJwtPubKeyMetric test flake

https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/31285/unit-tests_istio/1367946453724434432

See https://github.com/istio/istio/wiki/Test-Flakes for more info
cc @domcyrus

Able to reproduce figuring out how to solve...
```bash
~/go/bin/stress ./model.test -test.v -test.run TestJwtPubKeyMetric
5s: 52 runs so far, 0 failures
10s: 106 runs so far, 0 failures
15s: 161 runs so far, 0 failures
20s: 220 runs so far, 0 failures
25s: 276 runs so far, 0 failures
30s: 332 runs so far, 0 failures
35s: 379 runs so far, 0 failures
40s: 439 runs so far, 0 failures

/var/folders/9b/z1qg945j58xc8r8bhkx4ytn00000gp/T/go-stress-20210306T131828-360911413
=== RUN   TestJwtPubKeyMetric
2021-03-06T12:18:41.111839Z     info    Successfully serving on http://localhost:59170
2021-03-06T12:18:41.114723Z     error   model   Failed to fetch public key from "http://localhost:59170/oauth2/v3/certs": status 403, Mock server configured to return error until 1 retries
2021-03-06T12:18:41.159785Z     warn    model   Previous JWKs public key JSON is not parseable: 
2021-03-06T12:18:41.159885Z     info    model   Updated cached JWT public key from "http://localhost:59170/oauth2/v3/certs"
2021-03-06T12:18:41.212092Z     warn    model   Previous JWKs public key JSON is not parseable: 
2021-03-06T12:18:41.212166Z     info    model   Updated cached JWT public key from "http://localhost:59170/oauth2/v3/certs"
    jwks_resolver_test.go:494: retry.UntilOrFail: timeout while waiting after 24804 attempts (last error: expected condition not met)
--- FAIL: TestJwtPubKeyMetric (30.01s)
2021-03-06T12:19:11.115510Z     error   Server failed to serve in "http://localhost:59170": http: Server closed
FAIL

ERROR: exit status 1
```

--- a/pilot/pkg/model/jwks_resolver_test.go
+++ b/pilot/pkg/model/jwks_resolver_test.go
@@ -401,7 +401,7 @@ func TestJwtRefreshIntervalRecoverFromInitialFailOnFirstHit(t *testing.T) {
 
 	retry.UntilOrFail(t, func() bool {
 		pk, _ := r.GetPublicKey("", mockCertURL)
-		return test.JwtPubKey1 == pk
+		return test.JwtPubKey2 == pk
 	}, retry.Delay(time.Millisecond))
 	r.Close()
 
@@ -486,7 +486,7 @@ func TestJwtPubKeyMetric(t *testing.T) {
 		},
 		{
 			in:                []string{"", mockCertURL},
-			expectedJwtPubkey: test.JwtPubKey1,
+			expectedJwtPubkey: test.JwtPubKey2,
 		},
 	}
 

